### ISSUE TITLE
Subsoniq just crashes

### ISSUE THREAD

**Issue by KingDuckZ** [Znurre/Subsoniq#16]
I've been unable to pinpoint any actions I do, it seems to be mostly random (is there any multithreading going on, per chance?). As per the title, sometimes Subsoniq just crashes.
It might be unrelated, or a different crash even, but I noticed that sometimes the timeline on a song goes haywire, and it starts going forward by 20 seconds every few instants, looping back forever. During this process the music is just silent. If I click on another song sometimes playback would resume but most of the times the app will just crash the moment I click, or soon after.
This is especially annoying as playlists are not remembered, so after a crash one has to re-add all the songs that were in the queue before the crash.

**Comment by Znurre** [comment]
Thank you for the report :)
The Subsoniq library is multithreaded, although everything should be properly synced.
This project has been low priority for me the last few months, but I will make sure to have a look at the problem once I get coding on it again.
Your problem related to the looping is probably due to your encoder settings using variable bitrate.
This could also explain your other issue with the bitrate setting not working properly.
I will post a working encoder setting in the bitrate issue that you can try out and see if that works better for you.
Also, I can mention that persistent playlist support is one of the last things I added to the codebase, and I have it working on my own device.
If you want I could upload a build for you to try.

**Comment by KingDuckZ** [comment]
Thanks, but that's not a proper fix, the app should *not* crash. I managed to compile the desktop version of Subsoniq and it also segfaults miserably. I'll give it a go with the debugger attached next time, though I'm not an expert with Qt and stuff so I can't promise I'll find anything useful.

**Comment by KingDuckZ** [comment]
Speaking of the encoder part, I'm not sure what you mean there but the source ogg files are indeed VBR.

**Comment by Znurre** [comment]
Agreed, it should not crash. Any info you can find would be appreciated.
Please note that the desktop version is not close to production ready, and it might exhibit other bugs than the app :)
